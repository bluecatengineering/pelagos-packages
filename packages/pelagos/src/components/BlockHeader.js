import PropTypes from 'prop-types';
import {t} from '@bluecateng/l10n.macro';

import './BlockHeader.less';

/** Header for a form section, usually as a Collapsible title. */
const BlockHeader = ({className, header, configured}) => (
	<h2 className={`BlockHeader${className ? ` ${className}` : ''}`}>
		{header}
		<span className={`BlockHeader__indicator${configured ? ' BlockHeader--configured' : ''}`}>
			{configured ? t`(configured)` : t`(optional)`}
		</span>
	</h2>
);

BlockHeader.propTypes = {
	/** The component class name(s). */
	className: PropTypes.string,
	/** The header text. */
	header: PropTypes.string,
	/** Whether the value has been configured. */
	configured: PropTypes.bool,
};

export default BlockHeader;
