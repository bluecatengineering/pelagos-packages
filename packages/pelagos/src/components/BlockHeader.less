@import '../../less/fonts';
@import '../../less/spacing';

.BlockHeader {
	display: flex;
	flex-direction: row;
	align-items: center;
}

.BlockHeader__indicator {
	@font-10-400();
	color: var(--text-helper);
	margin-left: @sp-12;
}

.BlockHeader--configured {
	@font-12-400();
	color: var(--text-secondary);
}
