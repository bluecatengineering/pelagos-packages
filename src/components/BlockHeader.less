@import '../../less/fonts';
@import '../../less/spacing';

.BlockHeader {
	flex-direction: row;
	align-items: center;
}

.BlockHeader__indicator {
	@font-10-400();
	color: var(--font-color-3);
	margin-left: @sp-12;
}

.BlockHeader--configured {
	@font-12-400();
	color: var(--font-color-2);
}
