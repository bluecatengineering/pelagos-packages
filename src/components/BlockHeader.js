import PropTypes from 'prop-types';
import {t} from '@bluecat/l10n.macro';

import './BlockHeader.less';

const BlockHeader = ({header, configured}) => (
	<div className="BlockHeader">
		{header}
		<div className={'BlockHeader__indicator' + (configured ? ' BlockHeader--configured' : '')}>
			{configured ? t`(configured)` : t`(optional)`}
		</div>
	</div>
);

BlockHeader.propTypes = {
	header: PropTypes.string,
	configured: PropTypes.bool,
};

export default BlockHeader;
