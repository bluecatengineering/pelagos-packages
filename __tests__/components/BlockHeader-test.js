import {shallow} from 'enzyme';

import BlockHeader from '../../src/components/BlockHeader';

jest.unmock('../../src/components/BlockHeader');

describe('BlockHeader', () => {
	describe('rendering', () => {
		it('renders expected elements when configured is false', () => {
			const wrapper = shallow(<BlockHeader />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});

		it('renders expected elements when configured is true', () => {
			const wrapper = shallow(<BlockHeader configured={true} />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});
});
