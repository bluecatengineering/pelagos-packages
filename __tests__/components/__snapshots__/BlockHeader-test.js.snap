// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockHeader rendering renders expected elements when configured is false 1`] = `
<div
  className="BlockHeader"
>
  <div
    className="BlockHeader__indicator"
  >
    (optional)
  </div>
</div>
`;

exports[`BlockHeader rendering renders expected elements when configured is true 1`] = `
<div
  className="BlockHeader"
>
  <div
    className="BlockHeader__indicator BlockHeader--configured"
  >
    (configured)
  </div>
</div>
`;
